api/backend/app/services/agent_service.py

```diff
@@ -72,22 +74,37 @@ async def get_agent(self, agent_id: str) -> AgentResponseWithWalletDetails:
             agent_configurations=agent_configurations,
         )
 
-    async def update_agent(self, agent_id: str, agent_data: AgentUpdateDTO) -> AgentResponseWithAgentConfigurations:
+    async def update_agent(
+        self, agent_id: str, agent_data: AgentUpdateDTO, userAddress: str
+    ) -> AgentResponseWithAgentConfigurations:
+
         existing_agent = await self.agent_repository.retrieve_agent(agent_id)
         self.raise_exception_if_agent_not_found(existing_agent)
+        await self.is_authorized(userAddress, existing_agent)
+
+        user_is_super_admin = await self.user_reposiotry.is_super_admin(userAddress)
+        if existing_agent.user_address != userAddress and user_is_super_admin == False:
```

Comment:
this check is done twice in self.is_authorized and here.

Reply:
Removed repeated code.